src/document.rs

```diff
@@ -14,7 +15,12 @@ impl Document {
         let lang = ptr::null(); // Default language
 
         let doc = unsafe {
-            raw::RediSearch_CreateDocument(c_key.as_ptr() as *const c_void, key.len(), score, lang)
+            raw::RediSearch_CreateDocument(
+                c_key.as_ptr() as *const c_void,
+                key.len().try_into().unwrap(),
+                score,
+                lang,
+            )
```

Comment:
There's no need for this stuff, we can configure `bindgen` to map `usize` to `size_t` as it used to do before.
Simply apply this fix that I did to `redismodule-rs` (the `darwin` part is macOS specific, but does no harm on other platforms):

```
diff --git build.rs build.rs
index b194dca..f9bc442 100644
--- build.rs
+++ build.rs
@@ -13,6 +13,8 @@ fn main() {
         .whitelist_var("(RS|RediSearch|REDISEARCH_|GC_POLICY).*")
         .whitelist_function("RediSearch.*")
         .blacklist_item("RedisModule.*")
+        .blacklist_type("__darwin_.*")
+        .size_t_is_usize(true)
         .raw_line("use redis_module::raw::{RedisModuleCtx, RedisModuleString};")
         .generate()
         .expect("error generating RediSearch bindings");
```